tests/unit/test_gitlab_http_methods.py

```diff
@@ -189,218 +226,346 @@ def test_http_request_302_put_raises_redirect_error(gl):
+        gl.http_put("/not_there")
+    assert responses.assert_call_count(url, 1) is True
 
 
+@responses.activate
 def test_put_request_invalid_data(gl):
-    @urlmatch(scheme="http", netloc="localhost", path="/api/v4/projects", method="put")
-    def resp_cont(url, request):
-        headers = {"content-type": "application/json"}
-        content = '["name": "project1"]'
-        return response(200, content, headers, None, 5, request)
+    url = "http://localhost/api/v4/projects"
+    responses.add(
+        method=responses.PUT,
+        url=url,
+        body='["name": "project1"]',
+        content_type="application/json",
+        status=200,
+        match=[responses.matchers.query_param_matcher({})],
+    )
 
-    with HTTMock(resp_cont):
-        with pytest.raises(GitlabParsingError):
-            gl.http_put("/projects")
+    with pytest.raises(GitlabParsingError):
+        gl.http_put("/projects")
+    assert responses.assert_call_count(url, 1) is True
 
 
+@responses.activate
 def test_delete_request(gl):
-    @urlmatch(
-        scheme="http", netloc="localhost", path="/api/v4/projects", method="delete"
+    url = "http://localhost/api/v4/projects"
+    responses.add(
+        method=responses.DELETE,
+        url=url,
+        json=True,
+        status=200,
+        match=[responses.matchers.query_param_matcher({})],
     )
-    def resp_cont(url, request):
-        headers = {"content-type": "application/json"}
-        content = "true"
-        return response(200, content, headers, None, 5, request)
 
-    with HTTMock(resp_cont):
-        result = gl.http_delete("/projects")
-        assert isinstance(result, requests.Response)
-        assert result.json() is True
+    result = gl.http_delete("/projects")
+    assert isinstance(result, requests.Response)
+    assert result.json() is True
+    assert responses.assert_call_count(url, 1) is True
 
 
+@responses.activate
 def test_delete_request_404(gl):
-    @urlmatch(
-        scheme="http", netloc="localhost", path="/api/v4/not_there", method="delete"
+    url = "http://localhost/api/v4/not_there"
+    responses.add(
+        method=responses.DELETE,
+        url=url,
+        json=[],
+        status=404,
+        match=[responses.matchers.query_param_matcher({})],
     )
-    def resp_cont(url, request):
-        content = {"Here is why it failed"}
-        return response(404, content, {}, None, 5, request)
 
-    with HTTMock(resp_cont):
-        with pytest.raises(GitlabHttpError):
-            gl.http_delete("/not_there")
+    with pytest.raises(GitlabHttpError):
+        gl.http_delete("/not_there")
+    assert responses.assert_call_count(url, 1) is True
+
+
+# NOTE: The function `httmock_response` and the class `Headers` is taken from
+# https://github.com/patrys/httmock/ which is licensed under the Apache License, Version
+# 2.0. Thus it is allowed to be used in this project.
+# https://www.apache.org/licenses/GPL-compatibility.html
```

Comment:
Ah interesting, I assumed this would be possible without replicating httmock functionality, sorry for the detour then.

Maybe we could put this into a `utils.py` or `helpers.py` to separate it from test code, would also clean up the imports here. Same for the other helpers above that aren't fixtures perhaps.

Reply:
Will do.

Reply:
Done